### ISSUE TITLE
New Rule: disallow passing class methods as callbacks

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#783]
Every time you directly pass a class method as an argument to some function, you're probably doing it wrong. You want to bind using fat arrow instead.

```ts
// bad
someExternalLibrary.onClick(this.onClick)
```

```ts
// good
someExternalLibrary.onClick(() => this.onClick());
```

**Comment by markelliot** [comment]
dup of #259

**Comment by JKillian** [comment]
Good call, thanks @markelliot.
